src/components/structures/RoomView.js

```diff
@@ -677,7 +677,12 @@ module.exports = React.createClass({
 
     onRoomMemberMembership: function(ev, member, oldMembership) {
         if (member.userId == MatrixClientPeg.get().credentials.userId) {
-            this.forceUpdate();
```

Comment:
Do we not still need to do this forceUpdate in cases where where the membership !== join?